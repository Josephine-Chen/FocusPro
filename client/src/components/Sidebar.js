import React from 'react';
import getUserAjax from '../utils/UsersUtil';
import { getSettingAjax } from '../utils/SettingsUtil';
// export const Sidebar =({auth})=>(

export default class Sidebar extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      profile: this.props.auth.getProfile(),
      username: '',
      picture: '',
    };
    // listen to profile_updated events to update internal state
    this.props.auth.event.on('profile_updated', (newProfile) => {
      this.setState({ profile: newProfile }, () => {
        this.getUserInfo();
      });
    });
    this.getUserInfo = this.getUserInfo.bind(this);
  }

  componentDidMount() {
    this.getUserInfo();
  }

  getUserInfo() {
    getUserAjax(this.state.profile.user_id, (data)=>{
      this.setState({
        username: data.data[0].username || this.state.profile.nickname,
      });
    });
    getSettingAjax(this.state.profile.user_id, (data) => {
      this.setState({
        picture: data.picture || this.state.profile.picture,
      });
    });
  }

  render() {
    return (
      <header>
        <nav className="top-nav full hide-on-large-only">â€¨
          <a href="#" data-activates="slide-out" className="button-collapse top-nav full hide-on-large-only">
            <i className="material-icons">menu</i>
          </a>
        </nav>

        <ul id="slide-out" className="side-nav fixed theme-color">
          {localStorage.profile && <li><div className="userView">
            <div><img className="circle" src={this.state.picture} alt="auth0" /></div>
            <div><span className="name">{this.state.username}</span></div>
            <div><span className="email">{this.state.profile.email}</span></div>
            <a href="#" className="btn" onClick={this.props.auth.logout}>Log Out</a>
          </div></li>}

          <li>
            {!localStorage.id_token && <a href="#" className="btn login" onClick={this.props.auth.login}>Log In</a>}
          </li>
          {localStorage.id_token && <li><div className="divider" /></li>}
          {localStorage.id_token && <li><a href="#/dashboard" className="waves-effect"><i className="material-icons icon-white">dashboard</i>Dashboard</a></li>}
          {localStorage.id_token && <li><a href="#/settings" className="waves-effect"><i className="material-icons icon-white">settings</i>Settings</a></li>}
          {localStorage.id_token && <li><a href="#/selfreflection" className="waves-effect"><i className="material-icons icon-white">perm_identity</i>Self Reflections</a></li>}
        </ul>
      </header>
    );
  }
}
