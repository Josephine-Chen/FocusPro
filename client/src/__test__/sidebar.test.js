import React from 'react';
import renderer from 'react-test-renderer';
import {Sidebar} from '../components/Sidebar.js';
import { shallow, mount } from 'enzyme';

describe('Sidebar', () => {
  let sidebar;

  beforeAll(() => {
    sidebar = shallow(<Sidebar />);
  });

  // it('Sidebar renders title', () => {
  //   expect(sidebar.find('.page-title').text()).toEqual('FocusPro');
  // });
});

describe('Sidebar (Snapshot)', () => {
  it('Sidebar renders', () => {
    // const component = renderer.create(<Sidebar />);
    // const json = component.toJSON();
    // expect(json).toMatchSnapshot();
  });
});