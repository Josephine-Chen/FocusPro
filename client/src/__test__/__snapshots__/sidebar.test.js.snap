exports[`Sidebar (Snapshot) Sidebar renders 1`] = `
<header>
  <nav
    className="top-nav">
    <div
      className="container">
      <div
        className="nav-wrapper">
        <a
          className="page-title">
          FocusPro
        </a>
      </div>
    </div>
  </nav>
  <div
    className="container">
    <a
      className="button-collapse top-nav full hide-on-large-only"
      data-activates="slide-out"
      href="#">
      <i
        className="material-icons">
        menu
      </i>
    </a>
  </div>
  <ul
    className="side-nav fixed theme-color"
    id="slide-out">
    <li>
      <div
        className="userView">
        <a
          href="#">
          <img
            className="circle" />
        </a>
        <a
          href="#">
          <span
            className="name">
            John Doe
          </span>
        </a>
        <a
          href="#">
          <span
            className="email">
            jdandturk@gmail.com
          </span>
        </a>
        <a
          className="btn"
          href="#">
          Log Out
        </a>
      </div>
    </li>
    <li>
      <div
        className="divider" />
    </li>
    <li>
      <a
        className="waves-effect"
        href="#">
        <i
          className="material-icons icon-white">
          settings
        </i>
        Settings
      </a>
    </li>
    <li>
      <a
        className="waves-effect"
        href="#">
        <i
          className="material-icons icon-white">
          perm_identity
        </i>
        Self Reflections
      </a>
    </li>
  </ul>
</header>
`;
